yt_dlp/extractor/tunein.py

```diff
@@ -1,244 +1,332 @@
+import itertools
 import urllib.parse
 
 from .common import InfoExtractor
 from ..utils import (
-    OnDemandPagedList,
-    determine_ext,
+    clean_html,
+    int_or_none,
     parse_iso8601,
-    traverse_obj,
+    update_url_query,
+    url_or_none,
 )
+from ..utils.traversal import traverse_obj
 
 
 class TuneInBaseIE(InfoExtractor):
-    _VALID_URL_BASE = r'https?://(?:www\.)?tunein\.com'
+    def _call_api(self, item_id, endpoint=None, note='Downloading JSON metadata', query=None):
+        path = f'/{endpoint}' if endpoint else ''
 
-    def _extract_metadata(self, webpage, content_id):
-        return self._search_json(r'window.INITIAL_STATE=', webpage, 'hydration', content_id, fatal=False)
+        return self._download_json(
+            f'https://api.tunein.com/profiles/{item_id}{path}', item_id, note=note, query=query)
 
     def _extract_formats_and_subtitles(self, content_id):
         streams = self._download_json(
-            f'https://opml.radiotime.com/Tune.ashx?render=json&formats=mp3,aac,ogg,flash,hls&id={content_id}',
-            content_id)['body']
+            'https://opml.radiotime.com/Tune.ashx', content_id, query={
+                'formats': 'mp3,aac,ogg,flash,hls',
+                'id': content_id,
+                'render': 'json',
+            },
+        )['body']
```

Comment:
formatting, and let's traverse the response below
```suggestion
            })
```